gen3-integration-tests/services/gen3workflows.py

```diff
@@ -0,0 +1,133 @@
+from typing import Dict, Tuple
+
+import boto3
+import botocore.exceptions
+import pytest
+import requests
+from gen3.auth import Gen3Auth
+from utils import logger
+
+
+class Gen3Workflow:
+    def __init__(self):
+        self.BASE_URL = f"{pytest.root_url}"
+        self.SERVICE_URL = "/workflows"
+        self.S3_ENDPOINT_URL = f"{self.BASE_URL}{self.SERVICE_URL}/s3"
+
+    def _get_access_token(self, user: str = "main_account") -> str:
+        """Helper function to retrieve an access token."""
+        auth = Gen3Auth(refresh_token=pytest.api_keys[user], endpoint=self.BASE_URL)
+        return auth.get_access_token()
+
+    def _get_s3_client(self, access_token: str):
+        """Creates and returns an S3 client."""
+        return boto3.client(
+            service_name="s3",
+            aws_access_key_id=access_token,
+            aws_secret_access_key="N/A",  # pragma: allowlist secret
+            endpoint_url=self.S3_ENDPOINT_URL,
+        )
+
+    def _get_bucket_and_key(self, object_path: str) -> Tuple[str, str]:
+        """Parses object_path into bucket and key."""
+        # cleaner than using the string.split() method
+        bucket, _, key = object_path.partition("/")
+        return bucket, key
+
+    def get_storage_info(self, user: str = "main_account", expected_status=200) -> Dict:
+        """Makes a GET request to the `/storage/info` endpoint."""
+        access_token = self._get_access_token(user) if user else None
+        storage_url = f"{self.BASE_URL}{self.SERVICE_URL}/storage/info"
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"bearer {access_token}",
+        }
+
+        response = requests.get(url=storage_url, headers=headers)
+        assert (
+            response.status_code == expected_status
+        ), f"Expected {expected_status}, got {response.status_code}"
+
+        return response.json()
+
+    def get_bucket_object_with_unsigned_request(
+        self, object_path: str, user: str = "main_account", expected_status=401
+    ):
+        """Attempts to get an object without signing the request, expecting a failure."""
+        access_token = self._get_access_token(user)
+        s3_url = f"{self.S3_ENDPOINT_URL}/{object_path}"
+        headers = {"Authorization": f"bearer {access_token}"}
+
+        response = requests.get(url=s3_url, headers=headers)
+        assert (
+            response.status_code == expected_status
+        ), f"Expected {expected_status}, got {response.status_code}"
+
+    def perform_s3_action(
```

Comment:
```suggestion
    def _perform_s3_action(
```
since it's only used within the class